### PR TITLE
Modifying PDF URLs for papers to match Google Scholar redirects

### PR DESCRIPTION
**Pull request by arfon**
This PR changes the JOSS web application to point to PDFs local to the JOSS domain from the JOSS UI. This is a requirement for Google Scholar indexing:

> The PDF file for the article should be on the same host as the page with the metatags. Highlighted in green above. If this is fixed, would be best to update the citation_pdf_url metatag as well

/ cc #130

### CODE REVIEW

File: app/controllers/papers_controller.rb [hunk 1]
```diff
@@ -196,7 +196,10 @@ def show
       head 404 and return unless can_see_hidden_paper?(@paper)
     end
 
-    render layout: false
+    respond_to do |format|
+      format.pdf { redirect_to @paper.pdf_url, :status => 301 }
```
Comment (arfon) [inline app/controllers/papers_controller.rb]: If format is `pdf` we simply redirect to the existing `pdf_url` on GitHub pages.

File: app/models/paper.rb [hunk 2]
```diff
@@ -255,6 +255,26 @@ def joss_id
     "#{setting(:abbreviation).downcase}.#{id}"
   end
 
+  # This URL returns the 'DOI optimized' representation of a URL for a paper
+  # e.g. https://joss.theoj.org/papers/10.21105/joss.01632 rather than something
+  # with the SHA e.g. https://joss.theoj.org/papers/5e290cb57b61f83de4460fd0eca22726
+  # This URL format only works for accepted papers so falls back to the SHA
+  # version if no DOI is set.
+  def seo_url
```
Comment (arfon) [inline app/models/paper.rb]: There's probably a better way to do this with Rails routes and custom paths.

**Comment by arfon** [comment]
:wave: @xuanxu - this change works fine in Chrome and Firefox but Safari is giving me some kind of CORS error - perhaps it doesn't like the `301` for the PDF now?

Any ideas what might be going on here?

**Comment by xuanxu** [comment]
@arfon :thinking: I can't think of a reason for only failing in Safari. I'll try to test later today.

**Comment by arfon** [comment]
This feels related: https://github.com/NUKnightLab/StoryMapJS/issues/133

**Comment by arfon** [comment]
I just had the realization that I think the only URL that needs to point to the `https://joss.theo.org/.../blah.pdf` host is the Google Scholar meta tag (e.g. [this line](https://github.com/openjournals/joss/pull/605/files#diff-d426aa85f1b9364c3828995435a6aba7R62)) which has nothing to do with the embed view for the PDF.

So, I think I'm going to say this is fixed and that this is good enough for now.

**Comment by sayanddude** [comment]
@arfon Hi! I just had one question regarding indexing of my paper- https://joss.theoj.org/papers/10.21105/joss.01509 in Google scholar. The paper was published around 2 month back but it's not yet indexed in Google scholar whereas I see some recently published ones indexed in Google Scholar. Is there any way to fix this?

**Comment by arfon** [comment]
Good question! I'm not sure to be honest. The Google Scholar bot _eventually_ seems to pick up papers, why yours has not yet been indexed is a mystery sorry...

**Comment by sayanddude** [comment]
> Good question! I'm not sure to be honest. The Google Scholar bot _eventually_ seems to pick up papers, why yours has not yet been indexed is a mystery sorry...

Ok.... This is really strange!

**Pull request merged**